account_brand/views/res_partner_account_brand.xml

```diff
@@ -0,0 +1,32 @@
+<?xml version="1.0" encoding="utf-8"?>
+<!-- Copyright 2019 ACSONE SA/NV
+     License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl). -->
+
+<odoo>
+
+    <record model="ir.ui.view" id="res_partner_account_brand_tree_view">
+        <field name="name">res.partner.account.brand.tree (in account_brand)</field>
+        <field name="model">res.partner.account.brand</field>
+        <field name="arch" type="xml">
+            <tree editable="bottom">
+                <field name="brand_id"/>
+                <field name="account_type"/>
+                <field name="account_id"/>
```

Comment:
Add a dynamic domain on account_type.

Reply:
it's done in model level

https://github.com/OCA/brand/blob/1331850dba1f9802ad0939ef3f7ad49a93d14833/account_brand/models/res_partner_account_brand.py#L18-L23